[MINOR] release workflows corrected also added publish workflows

vedansh-5 opened the pull request:
### 📌 Fixes

Fixes #85 
---

### 📝 Summary of Changes

In this Pull Request I have added workflows that corrects the releases and draft notes, On each PR merge a new release will be drafted and tags will be updated based on semantic versioning and when we manually publish the release it will also be published to chrome web store.

---

### 📸 Screenshots / Demo (if UI-related)

_Add screenshots, video, or link to deployed preview if applicable_

---

### ✅ Checklist

- [ ] I’ve tested my changes locally
- [ ] I’ve added tests (if applicable)
- [ ] I’ve updated documentation (if applicable)
- [ ] My code follows the project’s code style guidelines

---

### 👀 Reviewer Notes

_Add any special notes for the reviewer here_

## Summary by Sourcery

Streamline release process by correcting release-drafter workflow to draft and tag new releases, automating CHANGELOG.md updates, and adding a publish workflow to deploy the extension to the Chrome Web Store

New Features:
- Add GitHub Actions workflow that zips and publishes the extension to the Chrome Web Store on manual release

Bug Fixes:
- Fix release-drafter triggers and configuration to properly draft releases and update tags on PR merges

Enhancements:
- Automate CHANGELOG.md generation and commit new entries for each release
- Refine release-drafter configuration with standardized templates and expanded maintenance labels

CI:
- Update release-drafter workflow to use release-drafter@v6 with publish flag, include checkout, Node.js setup, and changelog steps

vedansh-5 commented:
@hpdang This branch can be force-merged successfully, the failing job is a result of the current workflow, I have fixed the concerning code in this PR. thankyou

hpdang commented:
> Please add MAJOR, MINOR and PATCH in the titles of all the PRs that we have been merged or are still open, details on the same in readme changes in this PR. Thankyou :>

@vedansh-5 I think having the PR title with these labels are not so ideal. According to your README, I assume we can use GitHub label instead? I can create a new set of GitHub label. I am thinking to add also "release:none", so we will have the following labels:

- release:major
- release:minor
- release:patch
- release:none

And do we need to enforce PR labels on each single PR? I am also concerned that newcomers might not know which label is appropriate. What is your opinion?

hpdang commented:
And in the README, could you please also add a section explaining the release process?

vedansh-5 commented:
> @vedansh-5 I think having the PR title with these labels are not so ideal. According to your README, I assume we can use GitHub label instead? I can create a new set of GitHub label. I am thinking to add also "release:none", so we will have the following labels:
> 
> * release:major
> * release:minor
> * release:patch
> * release:none
> 
> And do we need to enforce PR labels on each single PR? I am also concerned that newcomers might not know which label is appropriate. What is your opinion?

I have made changes in the workflow to prioritize using the same labels you mentioned here and use the older MAJOR, MINOR, PATCH method as a fallback.
I have also added a section in the readme file for the new labels to give newcomers a better idea for labelling their work, in case the they fail to label it, we can label their work while reviewing, later on we can also add a workflow to comment on the PR for reminding them to add labels and directing the to the concerned section in the readme file.
I have added a section for the release process in the readme file.

vedansh-5 commented:
@hpdang I think adding a contributing guide for this project will be a good idea, I have created a draft for the same, lmk if you'd like to see it, I can raise a PR. Thankyou
